src/app/CommandHandler.cpp

```diff
@@ -97,6 +98,60 @@ void CommandHandler::OnInvokeCommandRequest(Messaging::ExchangeContext * ec, con
     mGoneAsync = true;
 }
 
+CHIP_ERROR CommandHandler::ValidateCommands(TLV::TLVReader & invokeRequestsReader)
+{
+    CHIP_ERROR err          = CHIP_NO_ERROR;
+    size_t commandCount     = 0;
+    bool commandRefExpected = false;
+
+    TLV::Utilities::Count(invokeRequestsReader, commandCount, false /* recurse */);
+    if (IsGroupRequest())
+    {
+        VerifyOrReturnError(commandCount == 1, CHIP_ERROR_INVALID_ARGUMENT);
+        return CHIP_NO_ERROR;
+    }
```

Comment:
Suggest early return `if (!IsGroupRequest())` (or else), that if commandCount == 1, it's automatically a success. This would allow removal of `commandRefExpected` (since it would always be implicitly true due to > 1), and those conditions depending on it could also be removed.

Reply:
I guess technically I have overloaded this function to server two purposes validate and save Command refs for future use. We still need to parse and save the CommandRef.

I will update the function name to be more clear that it servers these two purposes